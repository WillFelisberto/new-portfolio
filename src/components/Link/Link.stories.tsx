import { Meta, StoryFn } from '@storybook/react';
import Link from '.';

export default {
  title: 'Components/Link',
  component: Link,
  argTypes: {
    className: { control: 'text' },
    children: { control: 'text' },
    href: { control: 'text' },
  },
} satisfies Meta;

const Template: StoryFn<typeof Link> = (args) => <Link {...args} />;

export const Default = Template.bind({});
Default.args = {
  href: 'https://example.com',
  children: 'Default Link',
};

export const WithCustomClass = Template.bind({});
WithCustomClass.args = {
  href: 'https://example.com',
  children: 'Link with Custom Class',
  className: 'text-red-500',
};
