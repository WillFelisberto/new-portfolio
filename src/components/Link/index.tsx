import { cn } from '@/lib/utils';
import NextLink from 'next/link';
import { ComponentProps } from 'react';
type LinkProps = ComponentProps<typeof NextLink>;

const Link = ({ className, children, ...props }: LinkProps) => {
  return (
    <NextLink
      className={cn(
        'flex items-center gap-2 text-sm text-gray-300 transition-colors hover:text-sky-500',
        className,
      )}
      {...props}
    >
      {children}
    </NextLink>
  );
};

export default Link;
