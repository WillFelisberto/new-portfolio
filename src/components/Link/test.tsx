import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Link from '.';

describe('<Link />', () => {
  it('renders correctly with default props', () => {
    const { container } = render(
      <Link href="https://example.com">Default Link</Link>,
    );

    const link = screen.getByRole('link', { name: /default link/i });
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute('href', 'https://example.com');
    expect(link).toHaveClass(
      'flex items-center gap-2 text-sm text-gray-300 transition-colors hover:text-sky-500',
    );

    expect(container).toMatchSnapshot();
  });

  it('applies custom class names', () => {
    render(
      <Link href="https://example.com" className="custom-class">
        Custom Class Link
      </Link>,
    );

    const link = screen.getByRole('link', { name: /custom class link/i });
    expect(link).toHaveClass('custom-class');
  });

  it('renders children content', () => {
    render(<Link href="https://example.com">Child Content</Link>);

    const link = screen.getByRole('link', { name: /child content/i });
    expect(link).toBeInTheDocument();
    expect(link).toHaveTextContent('Child Content');
  });
});
