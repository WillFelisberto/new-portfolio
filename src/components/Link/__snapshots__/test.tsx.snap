// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> renders correctly with default props 1`] = `
<div>
  <a
    class="flex items-center gap-2 text-sm text-gray-300 transition-colors hover:text-sky-500"
    href="https://example.com"
  >
    Default Link
  </a>
</div>
`;
